src/realm/object-store/dictionary.cpp

```diff
@@ -346,7 +346,7 @@ class NotificationHandler {
 private:
     realm::Dictionary& m_dict;
     TransactionRef m_prev_rt;
-    std::unique_ptr<realm::Dictionary> m_prev_dict;
+    std::shared_ptr<realm::Dictionary> m_prev_dict;
```

Comment:
Why not to use `DictionaryPtr`?

Reply:
It is an error this was changed. It is a unique pointer that is returned from `import_copy_of`.
I will revert the change